ios/MaasCore/Sources/MaasComponents/Components/Button.swift

```diff
@@ -1,75 +1,159 @@
         _ text: String,
-        foreground: Color? = nil,
-        background: Color? = nil,
+        icon: Image? = nil,
+        foregroundColor: Color? = nil,
+        backgroundColor: Color? = nil,
+        isSmall: Bool = false,
+        isHugging: Bool = false,
+        isLoading: Binding<Bool> = .constant(false),
         action: @escaping () -> Void) {
-
+        
         input = InputType(
             text: text,
-            foreground: foreground,
-            background: background,
+            icon: icon,
+            foregroundColor: foregroundColor,
+            backgroundColor: backgroundColor,
+            isSmall: isSmall,
+            isHugging: isHugging,
+            isLoading: isLoading,
             action: action
         )
     }
-
-
+    
     @Environment(\.isEnabled) var isEnabled
-    @Themeable(ButtonConstants.init) var constants
-
+    @Themeable(ButtonConstants.init) var buttonConstants
+    
+    private var animation = Animation.linear(duration: 0.2)
+    
+    private var isLoading: Bool { input.isLoading.wrappedValue }
+    
     public var defaultBody: some View {
-        SwiftUI.Button(
-            action: input.action,
-            label: {
-                Text(input.text)
-                    .padding(.horizontal)
-                    .lineLimit(0)
-                    .minimumScaleFactor(0.75)
-                    .foregroundColor(input.foreground ?? (isEnabled ? constants.defaultContentColor : constants.disabledContentColor))
-                    .textStyle(constants.textStyle)
-                    .frame(maxWidth: .infinity, minHeight: constants.minHeight)
-                    .background(input.background ?? (isEnabled ? constants.defaultBackgroundColor : constants.disabledBackgroundColor))
-                    .cornerRadius(constants.cornerRadius)
+        SwiftUI.Button(action: input.action) {
+            HStack(spacing: buttonConstants.spaceBetween) {
+                
+                if isLoading || input.icon != nil {
+                    ZStack {
+                        
+                        if let icon = input.icon {
+                            image(icon)
+                        }
+                        
+                        spinner
+                    }
+                }
+                
+                if !input.text.isEmpty {
+                    text
+                }
             }
-        )
+            .padding(.horizontal, buttonConstants.paddingHorizontal)
+            .frame(
+                maxWidth: input.isHugging ? nil : .infinity,
+                minHeight: input.isSmall ? buttonConstants.heightSmall : buttonConstants.height
+            )
+            .animation(animation)
+        }
+        .foregroundColor(foregroundColor)
+        .background(backgroundColor)
+        .cornerRadius(buttonConstants.cornerRadius)
+        .animation(animation)
+    }
+    
+    private var text: some View {
+        Text(input.text)
+            .lineLimit(0)
```

Comment:
Does we really need a button with more than 1 line of text ? 🤔

Reply:
Probably no, left this from initial `Button` implementation